src/EnergyPlus/BaseboardRadiator.cc

```diff
@@ -843,9 +841,14 @@ namespace BaseboardRadiator {
                             SimHWConvective(state, baseboardNum, LoadMet);
 
                             if (LoadMet > DesCoilLoad) { // if the load met is greater than design load, OK to iterate on UA
-                                // Invert the baseboard model: given the design inlet conditions and the design load,
-                                // find the design UA.
-                                General::SolveRoot(state, Acc, MaxIte, SolFla, UA, HWBaseboardUAResidual, UA0, UA1, Par2);
+                                // Invert the baseboard model: given the design inlet conditions and the design load, find the design UA.
+                                auto f = [&state, &baseboardNum, &DesCoilLoad](Real64 UA) {
+                                    state.dataBaseboardRadiator->baseboards(baseboardNum).UA = UA;
+                                    Real64 LoadMet = 0.0;
+                                    SimHWConvective(state, baseboardNum, LoadMet);
+                                    return (DesCoilLoad - LoadMet) / DesCoilLoad;
+                                };
+                                General::SolveRoot(state, Acc, MaxIte, SolFla, UA, f, UA0, UA1);
```

Comment:
The new SolveRoot interface does not need a payload anymore; Par arrays are to be deleted.